### PR TITLE
[#45963] Remove Select All and Open Storage interaction elements from…

### PR DESCRIPTION
**Pull request by Kharonus**
… file pickers

- https://community.openproject.org/work_packages/45963
- introduced feature flag for select all feature
- removed open storage action button in modal header

### CODE REVIEW

File: frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts [hunk 1]
```diff
@@ -75,6 +75,10 @@ export class FilePickerModalComponent extends FilePickerBaseModalComponent {
     return this.selection.size;
   }
 
+  public get showSelectAll():boolean {
+    return this.configuration.activeFeatureFlags.includes('storageFilePickingSelectAll');
```
Comment (oliverguenther) [inline frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts]: Note that this getter will fire on every change detection, which is more costly than having a field evaluated once (e.g., in ngOnInit) that resolves to a boolean field.

File: frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts [hunk 2]
```diff
@@ -108,7 +113,7 @@ export class FilePickerModalComponent extends FilePickerBaseModalComponent {
       files,
     ).subscribe(
       (fileLinks) => { this.toastService.addSuccess(this.text.toast.successFileLinksCreated(fileLinks.count)); },
-      (error) => { this.toastService.addError(error); },
+      (error) => { this.toastService.addError(JSON.stringify(error)); },
```
Comment (oliverguenther) [inline frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts]: Do you really want to show a JSON to the user here?

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Some minor remarks / questions. Feel free to merge otherwise

File: frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts (hunk 1 above)
Comment (Kharonus) [reply frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts]: True, changed to static property filled in constructor.

wanted to rebase to test it with current dev, so no incremental diff.

File: frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts (hunk 2 above)
Comment (Kharonus) [reply frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts]: this is a typing issue here.

rxjs `error` in a subscriber callback is of type any. But our toast service only eats string, IToast or an HTTPErrorResponse.

Comment (Kharonus) [reply frontend/src/app/shared/components/storages/file-picker-modal/file-picker-modal.component.ts]: Solved as proposed with asserting the error to be of type `HttpErrorResponse`

**Pull request merged**